tests/models/plants/test_stoichiometry.py

```diff
@@ -0,0 +1,437 @@
+import numpy as np  # noqa: D100
+import pytest
+
+from virtual_ecosystem.models.plants.stoichiometry import (
+    FoliageTissue,
+    ReproductiveTissue,
+    RootTissue,
+    StemStoichiometry,
+    WoodTissue,
+)
+
+
+class DummyStemAllometry:
+    """Mock class for stem allometry with predefined masses for testing."""
+
+    foliage_mass = np.array([50.0, 80.0])
+    stem_mass = np.array([100.0, 150.0])
+    reproductive_tissue_mass = np.array([20.0, 40.0])
+
+
+class DummyStemTraits:
+    """Mock class for stem traits with predefined stoichiometric ratios for testing."""
+
+    zeta = 0.1
+    sla = 2.0
+    p_foliage_for_reproductive_tissue = 0.5
+
+
+class DummyExtraTraitsPFT:
+    """Minimal mock for extra PFT traits."""
+
+    pass
+
+
+class DummyCommunity:
+    """Mock class for a plant community with two cohorts for testing."""
+
+    def __init__(self):
+        self.stem_allometry = DummyStemAllometry()
+        self.stem_traits = DummyStemTraits()
```

Comment:
Fussy - but I'd be inclined to use `types.SimpleNamespace` to populate these values directly without having to create the two single use Dummy classes. Might be able to do the same with `DummyExtraTraitsPFT`?

```suggestion
        self.stem_allometry = SimpleNamespace(
            foliage_mass = np.array([50.0, 80.0]),
            stem_mass = np.array([100.0, 150.0]),
            reproductive_tissue_mass = np.array([20.0, 40.0])
        ),
        self.stem_traits = SimpleNamespace(
            zeta = 0.1, sla = 2.0, p_foliage_for_reproductive_tissue = 0.5
         )
        
```